pkg/istio-agent/xds_proxy.go

```diff
@@ -363,6 +374,44 @@ func (p *XdsProxy) handleUpstreamResponse(con *ProxyConnection) {
 					TypeUrl:       v3.NameTableType,
 					ResponseNonce: resp.Nonce,
 				}
+			case resource.ListenerType:
+				// intercept listeners, replace Wasm remote load with local file path.
+				// TODO: deal with fail open, or not?
+				// TODO: add parallel download
+				proxyLog.Info("intercept listener")
+				sendNack := false
+				for i, r := range resp.Resources {
+					var l listener.Listener
```

Comment:
This will delay init for all listeners even if only one of them has a wasm filter, that includes initial readiness. the key to consider here is ingress which may have multiple unrelated listeners.

It is not clear to me if responses are being streamed or batch processed here.

Reply:
Resources of the same type are batched in one response. I think that is how xds stow works?